### ISSUE TITLE
forUpdate - doesn't work

### ISSUE THREAD

**Issue by IAlexanderI** [phalcon/cphalcon#2407]
Hello All. 

I found bug in ORM, I think.

I use profiler to view all queries in page.
If I use pure sql query:

```php

$phql = "SELECT * FROM general_table where key = '$key' FOR UPDATE;";
$row = Phalcon\DI::getDefault()->getShared('db')->query($phql)->fetch();

```

logger showes to me - 'SELECT * FROM general_table where key = 'abc' FOR UPDATE;'

If I use model:

```php

$row = Model\General\GeneralTable::findFirst(
            array(
                "limit" => 1,
                "conditions" => 'key= :key:',
                "bind" => array('key'=>$key),
                "for_update" => true
            )
        );
		
```

logger showes only:  'SELECT * FROM general_table WHERE key = 'abc'

It's means "FOR UPDATE" doesn't append to query.

Query Builder also has same result.

Phalcon 1.3.2

**Comment by m1ome** [comment]
Yes, it's not fixed.

**Comment by m1ome** [comment]
@phalcon Please reopen this. It is not fixed

**Comment by anthmarques** [comment]
@phalcon not fixed

**Comment by banmaixanhjack** [comment]
Please fix this bug, we are developing an e-commerce based on Phalcon, and getting stuck because of this bug

**Comment by andresgutierrez** [comment]
Implemented in Phalcon 2.0.4

**Comment by ohmcoe** [comment]
shared_lock not working again in 2.0.9
